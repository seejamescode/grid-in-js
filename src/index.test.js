import gridWiz from './index';

const defaultConfig = {
  prefix: "bx--",
  support: "displayGrid",
  maxWidth: 1584,
  progressive: false,
  subgrid: true,
  breakpoints: [
    {
      name: "sm",
      size: 0,
      columns: 4,
      gutter: 32,
      margin: 0
    },
    {
      name: "md",
      size: 672,
      columns: 8,
      gutter: 32,
      margin: 16
    },
    {
      name: "lg",
      size: 1056,
      columns: 16,
      gutter: 32,
      margin: 16
    },
    {
      name: "xl",
      size: 1312,
      columns: 16,
      gutter: 32,
      margin: 16
    },
    {
      name: "max",
      size: 1584,
      columns: 16,
      gutter: 32,
      margin: 32
    }
  ]
};

describe('gridWiz', () => {
  it('should generate css according to default config', () => {
    expect(gridWiz(defaultConfig)).toMatchSnapshot();
  });
});
