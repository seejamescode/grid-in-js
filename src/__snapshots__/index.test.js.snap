// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gridWiz should generate css according to default config 1`] = `" body { margin: 0; } .bx--grid, .bx--grid * { box-sizing: border-box; } .bx--grid, .bx--row { --bx--x-col-columns: 1; display: grid; } .bx--grid .bx--grid, .bx--grid .bx--row { margin-left: 0; margin-right: 0; padding-left: 0; padding-right: 0; } .bx--grid { padding-left: 0px; padding-right: 0px; } @media (min-width: 672px) { .bx--grid { padding-left: 16px; padding-right: 16px; } }@media (min-width: 1584px) { .bx--grid { padding-left: 32px; padding-right: 32px; } } @media (min-width: 1648px) { .bx--grid { margin-left: auto; margin-right: auto; max-width: 1584px; } } .bx--grid > *, .bx--row > * { grid-column-end: span var(--bx--x-bp-columns); } [class*=\\"bx--col\\"]:not(.bx--grid):not(.bx--row) { padding-left: var(--bx--x-gutter); padding-right: var(--bx--x-gutter); } .bx--grid, .bx--row { grid-template-columns: repeat(auto-fill, calc(25% / var(--bx--x-col-columns))); --bx--x-bp-columns: 4; } [class*=\\"bx--col-sm\\"] { display: initial; } [class*=\\"bx--col-sm\\"].bx--grid, [class*=\\"bx--col-sm\\"].bx--row { display: grid; } .bx--col-sm-0, .bx--col-sm-0.bx--grid, .bx--col-sm-0.bx--row { display: none; } .bx--col-sm-1{ grid-column-end: span var(--bx--x-bp); } .bx--grid .bx--offset-sm-1, .bx--row .bx--offset-sm-1 { grid-column-start: calc(var(--bx--x-bp) + 1); } .bx--col-sm-2{ grid-column-end: span calc(2 * var(--bx--x-bp)); } .bx--grid .bx--offset-sm-2, .bx--row .bx--offset-sm-2 { grid-column-start: calc(2 * var(--bx--x-bp) + 1); } .bx--col-sm-3{ grid-column-end: span calc(3 * var(--bx--x-bp)); } .bx--grid .bx--offset-sm-3, .bx--row .bx--offset-sm-3 { grid-column-start: calc(3 * var(--bx--x-bp) + 1); } .bx--col-sm-4{ grid-column-end: span calc(4 * var(--bx--x-bp)); } [class*=\\"bx--col-sm\\"] { --bx--x-bp: 1; } :root { --bx--x-gutter: 16px; } .bx--col-sm-1 { --bx--x-col-columns: 0.25; } .bx--col-sm-2 { --bx--x-col-columns: 0.5; } .bx--col-sm-3 { --bx--x-col-columns: 0.75; } @media (min-width: 672px) { .bx--grid, .bx--row { grid-template-columns: repeat(auto-fill, calc(12.5% / var(--bx--x-col-columns))); --bx--x-bp-columns: 8; } [class*=\\"bx--col-md\\"] { display: initial; } [class*=\\"bx--col-md\\"].bx--grid, [class*=\\"bx--col-md\\"].bx--row { display: grid; } .bx--col-md-0, .bx--col-md-0.bx--grid, .bx--col-md-0.bx--row { display: none; } .bx--col-md-1{ grid-column-end: span var(--bx--x-bp); } .bx--grid .bx--offset-md-1, .bx--row .bx--offset-md-1 { grid-column-start: calc(var(--bx--x-bp) + 1); } .bx--col-md-2{ grid-column-end: span calc(2 * var(--bx--x-bp)); } .bx--grid .bx--offset-md-2, .bx--row .bx--offset-md-2 { grid-column-start: calc(2 * var(--bx--x-bp) + 1); } .bx--col-md-3{ grid-column-end: span calc(3 * var(--bx--x-bp)); } .bx--grid .bx--offset-md-3, .bx--row .bx--offset-md-3 { grid-column-start: calc(3 * var(--bx--x-bp) + 1); } .bx--col-md-4{ grid-column-end: span calc(4 * var(--bx--x-bp)); } .bx--grid .bx--offset-md-4, .bx--row .bx--offset-md-4 { grid-column-start: calc(4 * var(--bx--x-bp) + 1); } .bx--col-md-5{ grid-column-end: span calc(5 * var(--bx--x-bp)); } .bx--grid .bx--offset-md-5, .bx--row .bx--offset-md-5 { grid-column-start: calc(5 * var(--bx--x-bp) + 1); } .bx--col-md-6{ grid-column-end: span calc(6 * var(--bx--x-bp)); } .bx--grid .bx--offset-md-6, .bx--row .bx--offset-md-6 { grid-column-start: calc(6 * var(--bx--x-bp) + 1); } .bx--col-md-7{ grid-column-end: span calc(7 * var(--bx--x-bp)); } .bx--grid .bx--offset-md-7, .bx--row .bx--offset-md-7 { grid-column-start: calc(7 * var(--bx--x-bp) + 1); } .bx--col-md-8{ grid-column-end: span calc(8 * var(--bx--x-bp)); } [class*=\\"bx--col-sm\\"] { --bx--x-bp: 2; } :root { --bx--x-gutter: 16px; } .bx--col-sm-1 { --bx--x-col-columns: 0.25; } .bx--col-sm-2 { --bx--x-col-columns: 0.5; } .bx--col-sm-3 { --bx--x-col-columns: 0.75; } [class*=\\"bx--col-md\\"] { --bx--x-bp: 1; } .bx--col-md-1 { --bx--x-col-columns: 0.125; } .bx--col-md-2 { --bx--x-col-columns: 0.25; } .bx--col-md-3 { --bx--x-col-columns: 0.375; } .bx--col-md-4 { --bx--x-col-columns: 0.5; } .bx--col-md-5 { --bx--x-col-columns: 0.625; } .bx--col-md-6 { --bx--x-col-columns: 0.75; } .bx--col-md-7 { --bx--x-col-columns: 0.875; } }@media (min-width: 1056px) { .bx--grid, .bx--row { grid-template-columns: repeat(auto-fill, calc(6.25% / var(--bx--x-col-columns))); --bx--x-bp-columns: 16; } [class*=\\"bx--col-lg\\"] { display: initial; } [class*=\\"bx--col-lg\\"].bx--grid, [class*=\\"bx--col-lg\\"].bx--row { display: grid; } .bx--col-lg-0, .bx--col-lg-0.bx--grid, .bx--col-lg-0.bx--row { display: none; } .bx--col-lg-1{ grid-column-end: span var(--bx--x-bp); } .bx--grid .bx--offset-lg-1, .bx--row .bx--offset-lg-1 { grid-column-start: calc(var(--bx--x-bp) + 1); } .bx--col-lg-2{ grid-column-end: span calc(2 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-2, .bx--row .bx--offset-lg-2 { grid-column-start: calc(2 * var(--bx--x-bp) + 1); } .bx--col-lg-3{ grid-column-end: span calc(3 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-3, .bx--row .bx--offset-lg-3 { grid-column-start: calc(3 * var(--bx--x-bp) + 1); } .bx--col-lg-4{ grid-column-end: span calc(4 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-4, .bx--row .bx--offset-lg-4 { grid-column-start: calc(4 * var(--bx--x-bp) + 1); } .bx--col-lg-5{ grid-column-end: span calc(5 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-5, .bx--row .bx--offset-lg-5 { grid-column-start: calc(5 * var(--bx--x-bp) + 1); } .bx--col-lg-6{ grid-column-end: span calc(6 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-6, .bx--row .bx--offset-lg-6 { grid-column-start: calc(6 * var(--bx--x-bp) + 1); } .bx--col-lg-7{ grid-column-end: span calc(7 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-7, .bx--row .bx--offset-lg-7 { grid-column-start: calc(7 * var(--bx--x-bp) + 1); } .bx--col-lg-8{ grid-column-end: span calc(8 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-8, .bx--row .bx--offset-lg-8 { grid-column-start: calc(8 * var(--bx--x-bp) + 1); } .bx--col-lg-9{ grid-column-end: span calc(9 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-9, .bx--row .bx--offset-lg-9 { grid-column-start: calc(9 * var(--bx--x-bp) + 1); } .bx--col-lg-10{ grid-column-end: span calc(10 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-10, .bx--row .bx--offset-lg-10 { grid-column-start: calc(10 * var(--bx--x-bp) + 1); } .bx--col-lg-11{ grid-column-end: span calc(11 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-11, .bx--row .bx--offset-lg-11 { grid-column-start: calc(11 * var(--bx--x-bp) + 1); } .bx--col-lg-12{ grid-column-end: span calc(12 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-12, .bx--row .bx--offset-lg-12 { grid-column-start: calc(12 * var(--bx--x-bp) + 1); } .bx--col-lg-13{ grid-column-end: span calc(13 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-13, .bx--row .bx--offset-lg-13 { grid-column-start: calc(13 * var(--bx--x-bp) + 1); } .bx--col-lg-14{ grid-column-end: span calc(14 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-14, .bx--row .bx--offset-lg-14 { grid-column-start: calc(14 * var(--bx--x-bp) + 1); } .bx--col-lg-15{ grid-column-end: span calc(15 * var(--bx--x-bp)); } .bx--grid .bx--offset-lg-15, .bx--row .bx--offset-lg-15 { grid-column-start: calc(15 * var(--bx--x-bp) + 1); } .bx--col-lg-16{ grid-column-end: span calc(16 * var(--bx--x-bp)); } [class*=\\"bx--col-sm\\"] { --bx--x-bp: 4; } :root { --bx--x-gutter: 16px; } .bx--col-sm-1 { --bx--x-col-columns: 0.25; } .bx--col-sm-2 { --bx--x-col-columns: 0.5; } .bx--col-sm-3 { --bx--x-col-columns: 0.75; } [class*=\\"bx--col-md\\"] { --bx--x-bp: 2; } .bx--col-md-1 { --bx--x-col-columns: 0.125; } .bx--col-md-2 { --bx--x-col-columns: 0.25; } .bx--col-md-3 { --bx--x-col-columns: 0.375; } .bx--col-md-4 { --bx--x-col-columns: 0.5; } .bx--col-md-5 { --bx--x-col-columns: 0.625; } .bx--col-md-6 { --bx--x-col-columns: 0.75; } .bx--col-md-7 { --bx--x-col-columns: 0.875; } [class*=\\"bx--col-lg\\"] { --bx--x-bp: 1; } .bx--col-lg-1 { --bx--x-col-columns: 0.0625; } .bx--col-lg-2 { --bx--x-col-columns: 0.125; } .bx--col-lg-3 { --bx--x-col-columns: 0.1875; } .bx--col-lg-4 { --bx--x-col-columns: 0.25; } .bx--col-lg-5 { --bx--x-col-columns: 0.3125; } .bx--col-lg-6 { --bx--x-col-columns: 0.375; } .bx--col-lg-7 { --bx--x-col-columns: 0.4375; } .bx--col-lg-8 { --bx--x-col-columns: 0.5; } .bx--col-lg-9 { --bx--x-col-columns: 0.5625; } .bx--col-lg-10 { --bx--x-col-columns: 0.625; } .bx--col-lg-11 { --bx--x-col-columns: 0.6875; } .bx--col-lg-12 { --bx--x-col-columns: 0.75; } .bx--col-lg-13 { --bx--x-col-columns: 0.8125; } .bx--col-lg-14 { --bx--x-col-columns: 0.875; } .bx--col-lg-15 { --bx--x-col-columns: 0.9375; } }@media (min-width: 1312px) { .bx--grid, .bx--row { grid-template-columns: repeat(auto-fill, calc(6.25% / var(--bx--x-col-columns))); --bx--x-bp-columns: 16; } [class*=\\"bx--col-xl\\"] { display: initial; } [class*=\\"bx--col-xl\\"].bx--grid, [class*=\\"bx--col-xl\\"].bx--row { display: grid; } .bx--col-xl-0, .bx--col-xl-0.bx--grid, .bx--col-xl-0.bx--row { display: none; } .bx--col-xl-1{ grid-column-end: span var(--bx--x-bp); } .bx--grid .bx--offset-xl-1, .bx--row .bx--offset-xl-1 { grid-column-start: calc(var(--bx--x-bp) + 1); } .bx--col-xl-2{ grid-column-end: span calc(2 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-2, .bx--row .bx--offset-xl-2 { grid-column-start: calc(2 * var(--bx--x-bp) + 1); } .bx--col-xl-3{ grid-column-end: span calc(3 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-3, .bx--row .bx--offset-xl-3 { grid-column-start: calc(3 * var(--bx--x-bp) + 1); } .bx--col-xl-4{ grid-column-end: span calc(4 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-4, .bx--row .bx--offset-xl-4 { grid-column-start: calc(4 * var(--bx--x-bp) + 1); } .bx--col-xl-5{ grid-column-end: span calc(5 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-5, .bx--row .bx--offset-xl-5 { grid-column-start: calc(5 * var(--bx--x-bp) + 1); } .bx--col-xl-6{ grid-column-end: span calc(6 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-6, .bx--row .bx--offset-xl-6 { grid-column-start: calc(6 * var(--bx--x-bp) + 1); } .bx--col-xl-7{ grid-column-end: span calc(7 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-7, .bx--row .bx--offset-xl-7 { grid-column-start: calc(7 * var(--bx--x-bp) + 1); } .bx--col-xl-8{ grid-column-end: span calc(8 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-8, .bx--row .bx--offset-xl-8 { grid-column-start: calc(8 * var(--bx--x-bp) + 1); } .bx--col-xl-9{ grid-column-end: span calc(9 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-9, .bx--row .bx--offset-xl-9 { grid-column-start: calc(9 * var(--bx--x-bp) + 1); } .bx--col-xl-10{ grid-column-end: span calc(10 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-10, .bx--row .bx--offset-xl-10 { grid-column-start: calc(10 * var(--bx--x-bp) + 1); } .bx--col-xl-11{ grid-column-end: span calc(11 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-11, .bx--row .bx--offset-xl-11 { grid-column-start: calc(11 * var(--bx--x-bp) + 1); } .bx--col-xl-12{ grid-column-end: span calc(12 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-12, .bx--row .bx--offset-xl-12 { grid-column-start: calc(12 * var(--bx--x-bp) + 1); } .bx--col-xl-13{ grid-column-end: span calc(13 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-13, .bx--row .bx--offset-xl-13 { grid-column-start: calc(13 * var(--bx--x-bp) + 1); } .bx--col-xl-14{ grid-column-end: span calc(14 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-14, .bx--row .bx--offset-xl-14 { grid-column-start: calc(14 * var(--bx--x-bp) + 1); } .bx--col-xl-15{ grid-column-end: span calc(15 * var(--bx--x-bp)); } .bx--grid .bx--offset-xl-15, .bx--row .bx--offset-xl-15 { grid-column-start: calc(15 * var(--bx--x-bp) + 1); } .bx--col-xl-16{ grid-column-end: span calc(16 * var(--bx--x-bp)); } [class*=\\"bx--col-sm\\"] { --bx--x-bp: 4; } :root { --bx--x-gutter: 16px; } .bx--col-sm-1 { --bx--x-col-columns: 0.25; } .bx--col-sm-2 { --bx--x-col-columns: 0.5; } .bx--col-sm-3 { --bx--x-col-columns: 0.75; } [class*=\\"bx--col-md\\"] { --bx--x-bp: 2; } .bx--col-md-1 { --bx--x-col-columns: 0.125; } .bx--col-md-2 { --bx--x-col-columns: 0.25; } .bx--col-md-3 { --bx--x-col-columns: 0.375; } .bx--col-md-4 { --bx--x-col-columns: 0.5; } .bx--col-md-5 { --bx--x-col-columns: 0.625; } .bx--col-md-6 { --bx--x-col-columns: 0.75; } .bx--col-md-7 { --bx--x-col-columns: 0.875; } [class*=\\"bx--col-lg\\"] { --bx--x-bp: 1; } .bx--col-lg-1 { --bx--x-col-columns: 0.0625; } .bx--col-lg-2 { --bx--x-col-columns: 0.125; } .bx--col-lg-3 { --bx--x-col-columns: 0.1875; } .bx--col-lg-4 { --bx--x-col-columns: 0.25; } .bx--col-lg-5 { --bx--x-col-columns: 0.3125; } .bx--col-lg-6 { --bx--x-col-columns: 0.375; } .bx--col-lg-7 { --bx--x-col-columns: 0.4375; } .bx--col-lg-8 { --bx--x-col-columns: 0.5; } .bx--col-lg-9 { --bx--x-col-columns: 0.5625; } .bx--col-lg-10 { --bx--x-col-columns: 0.625; } .bx--col-lg-11 { --bx--x-col-columns: 0.6875; } .bx--col-lg-12 { --bx--x-col-columns: 0.75; } .bx--col-lg-13 { --bx--x-col-columns: 0.8125; } .bx--col-lg-14 { --bx--x-col-columns: 0.875; } .bx--col-lg-15 { --bx--x-col-columns: 0.9375; } [class*=\\"bx--col-xl\\"] { --bx--x-bp: 1; } .bx--col-xl-1 { --bx--x-col-columns: 0.0625; } .bx--col-xl-2 { --bx--x-col-columns: 0.125; } .bx--col-xl-3 { --bx--x-col-columns: 0.1875; } .bx--col-xl-4 { --bx--x-col-columns: 0.25; } .bx--col-xl-5 { --bx--x-col-columns: 0.3125; } .bx--col-xl-6 { --bx--x-col-columns: 0.375; } .bx--col-xl-7 { --bx--x-col-columns: 0.4375; } .bx--col-xl-8 { --bx--x-col-columns: 0.5; } .bx--col-xl-9 { --bx--x-col-columns: 0.5625; } .bx--col-xl-10 { --bx--x-col-columns: 0.625; } .bx--col-xl-11 { --bx--x-col-columns: 0.6875; } .bx--col-xl-12 { --bx--x-col-columns: 0.75; } .bx--col-xl-13 { --bx--x-col-columns: 0.8125; } .bx--col-xl-14 { --bx--x-col-columns: 0.875; } .bx--col-xl-15 { --bx--x-col-columns: 0.9375; } }@media (min-width: 1584px) { .bx--grid, .bx--row { grid-template-columns: repeat(auto-fill, calc(6.25% / var(--bx--x-col-columns))); --bx--x-bp-columns: 16; } [class*=\\"bx--col-max\\"] { display: initial; } [class*=\\"bx--col-max\\"].bx--grid, [class*=\\"bx--col-max\\"].bx--row { display: grid; } .bx--col-max-0, .bx--col-max-0.bx--grid, .bx--col-max-0.bx--row { display: none; } .bx--col-max-1{ grid-column-end: span var(--bx--x-bp); } .bx--grid .bx--offset-max-1, .bx--row .bx--offset-max-1 { grid-column-start: calc(var(--bx--x-bp) + 1); } .bx--col-max-2{ grid-column-end: span calc(2 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-2, .bx--row .bx--offset-max-2 { grid-column-start: calc(2 * var(--bx--x-bp) + 1); } .bx--col-max-3{ grid-column-end: span calc(3 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-3, .bx--row .bx--offset-max-3 { grid-column-start: calc(3 * var(--bx--x-bp) + 1); } .bx--col-max-4{ grid-column-end: span calc(4 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-4, .bx--row .bx--offset-max-4 { grid-column-start: calc(4 * var(--bx--x-bp) + 1); } .bx--col-max-5{ grid-column-end: span calc(5 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-5, .bx--row .bx--offset-max-5 { grid-column-start: calc(5 * var(--bx--x-bp) + 1); } .bx--col-max-6{ grid-column-end: span calc(6 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-6, .bx--row .bx--offset-max-6 { grid-column-start: calc(6 * var(--bx--x-bp) + 1); } .bx--col-max-7{ grid-column-end: span calc(7 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-7, .bx--row .bx--offset-max-7 { grid-column-start: calc(7 * var(--bx--x-bp) + 1); } .bx--col-max-8{ grid-column-end: span calc(8 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-8, .bx--row .bx--offset-max-8 { grid-column-start: calc(8 * var(--bx--x-bp) + 1); } .bx--col-max-9{ grid-column-end: span calc(9 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-9, .bx--row .bx--offset-max-9 { grid-column-start: calc(9 * var(--bx--x-bp) + 1); } .bx--col-max-10{ grid-column-end: span calc(10 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-10, .bx--row .bx--offset-max-10 { grid-column-start: calc(10 * var(--bx--x-bp) + 1); } .bx--col-max-11{ grid-column-end: span calc(11 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-11, .bx--row .bx--offset-max-11 { grid-column-start: calc(11 * var(--bx--x-bp) + 1); } .bx--col-max-12{ grid-column-end: span calc(12 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-12, .bx--row .bx--offset-max-12 { grid-column-start: calc(12 * var(--bx--x-bp) + 1); } .bx--col-max-13{ grid-column-end: span calc(13 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-13, .bx--row .bx--offset-max-13 { grid-column-start: calc(13 * var(--bx--x-bp) + 1); } .bx--col-max-14{ grid-column-end: span calc(14 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-14, .bx--row .bx--offset-max-14 { grid-column-start: calc(14 * var(--bx--x-bp) + 1); } .bx--col-max-15{ grid-column-end: span calc(15 * var(--bx--x-bp)); } .bx--grid .bx--offset-max-15, .bx--row .bx--offset-max-15 { grid-column-start: calc(15 * var(--bx--x-bp) + 1); } .bx--col-max-16{ grid-column-end: span calc(16 * var(--bx--x-bp)); } [class*=\\"bx--col-sm\\"] { --bx--x-bp: 4; } :root { --bx--x-gutter: 16px; } .bx--col-sm-1 { --bx--x-col-columns: 0.25; } .bx--col-sm-2 { --bx--x-col-columns: 0.5; } .bx--col-sm-3 { --bx--x-col-columns: 0.75; } [class*=\\"bx--col-md\\"] { --bx--x-bp: 2; } .bx--col-md-1 { --bx--x-col-columns: 0.125; } .bx--col-md-2 { --bx--x-col-columns: 0.25; } .bx--col-md-3 { --bx--x-col-columns: 0.375; } .bx--col-md-4 { --bx--x-col-columns: 0.5; } .bx--col-md-5 { --bx--x-col-columns: 0.625; } .bx--col-md-6 { --bx--x-col-columns: 0.75; } .bx--col-md-7 { --bx--x-col-columns: 0.875; } [class*=\\"bx--col-lg\\"] { --bx--x-bp: 1; } .bx--col-lg-1 { --bx--x-col-columns: 0.0625; } .bx--col-lg-2 { --bx--x-col-columns: 0.125; } .bx--col-lg-3 { --bx--x-col-columns: 0.1875; } .bx--col-lg-4 { --bx--x-col-columns: 0.25; } .bx--col-lg-5 { --bx--x-col-columns: 0.3125; } .bx--col-lg-6 { --bx--x-col-columns: 0.375; } .bx--col-lg-7 { --bx--x-col-columns: 0.4375; } .bx--col-lg-8 { --bx--x-col-columns: 0.5; } .bx--col-lg-9 { --bx--x-col-columns: 0.5625; } .bx--col-lg-10 { --bx--x-col-columns: 0.625; } .bx--col-lg-11 { --bx--x-col-columns: 0.6875; } .bx--col-lg-12 { --bx--x-col-columns: 0.75; } .bx--col-lg-13 { --bx--x-col-columns: 0.8125; } .bx--col-lg-14 { --bx--x-col-columns: 0.875; } .bx--col-lg-15 { --bx--x-col-columns: 0.9375; } [class*=\\"bx--col-xl\\"] { --bx--x-bp: 1; } .bx--col-xl-1 { --bx--x-col-columns: 0.0625; } .bx--col-xl-2 { --bx--x-col-columns: 0.125; } .bx--col-xl-3 { --bx--x-col-columns: 0.1875; } .bx--col-xl-4 { --bx--x-col-columns: 0.25; } .bx--col-xl-5 { --bx--x-col-columns: 0.3125; } .bx--col-xl-6 { --bx--x-col-columns: 0.375; } .bx--col-xl-7 { --bx--x-col-columns: 0.4375; } .bx--col-xl-8 { --bx--x-col-columns: 0.5; } .bx--col-xl-9 { --bx--x-col-columns: 0.5625; } .bx--col-xl-10 { --bx--x-col-columns: 0.625; } .bx--col-xl-11 { --bx--x-col-columns: 0.6875; } .bx--col-xl-12 { --bx--x-col-columns: 0.75; } .bx--col-xl-13 { --bx--x-col-columns: 0.8125; } .bx--col-xl-14 { --bx--x-col-columns: 0.875; } .bx--col-xl-15 { --bx--x-col-columns: 0.9375; } [class*=\\"bx--col-max\\"] { --bx--x-bp: 1; } .bx--col-max-1 { --bx--x-col-columns: 0.0625; } .bx--col-max-2 { --bx--x-col-columns: 0.125; } .bx--col-max-3 { --bx--x-col-columns: 0.1875; } .bx--col-max-4 { --bx--x-col-columns: 0.25; } .bx--col-max-5 { --bx--x-col-columns: 0.3125; } .bx--col-max-6 { --bx--x-col-columns: 0.375; } .bx--col-max-7 { --bx--x-col-columns: 0.4375; } .bx--col-max-8 { --bx--x-col-columns: 0.5; } .bx--col-max-9 { --bx--x-col-columns: 0.5625; } .bx--col-max-10 { --bx--x-col-columns: 0.625; } .bx--col-max-11 { --bx--x-col-columns: 0.6875; } .bx--col-max-12 { --bx--x-col-columns: 0.75; } .bx--col-max-13 { --bx--x-col-columns: 0.8125; } .bx--col-max-14 { --bx--x-col-columns: 0.875; } .bx--col-max-15 { --bx--x-col-columns: 0.9375; } }"`;
